src/Components/Components/test/CascadingParameterStateTest.cs

```diff
@@ -385,7 +386,7 @@ static CascadingValue<T> CreateCascadingValueComponent<T>(T value, string name =
                 supplierParams.Add("Name", name);
             }
 
-            supplier.SetParameters(supplierParams);
+            renderer.Invoke(() => supplier.SetParameters(supplierParams));
```

Comment:
All code that triggers renders needs to execute in the context of the render, but that doesn't change the behavior of the code with regards to synchronicity (that's why we don't have to await things here)

Reply:
I removed all these by putting them inside the testcomponent and testrenderer, it's the same, just cleaner. I take care of asserting that everything happens synchronously and unwrap and rethrow exceptions as appropriate